src/colmap/sensor/models.h

```diff
@@ -960,41 +1031,60 @@ std::vector<double> OpenCVFisheyeCameraModel::InitializeParams(
 }
 
 template <typename T>
-void OpenCVFisheyeCameraModel::ImgFromCam(
-    const T* params, T u, T v, T w, T* x, T* y) {
+void OpenCVFisheyeCameraModel::ImgFromFisheyeCoordinate(
+    const T* params, T uu, T vv, T* x, T* y) {
+  const T f1 = params[0];
+  const T f2 = params[1];
+  const T c1 = params[2];
+  const T c2 = params[3];
+
+  *x = f1 * uu + c1;
+  *y = f2 * vv + c2;
+}
+
+template <typename T>
+void OpenCVFisheyeCameraModel::FisheyeCoordinateFromImg(
+    const T* params, T x, T y, T* uu, T* vv) {
   const T f1 = params[0];
   const T f2 = params[1];
   const T c1 = params[2];
   const T c2 = params[3];
 
+  *uu = (x - c1) / f1;
+  *vv = (y - c2) / f2;
+}
+
+template <typename T>
+void OpenCVFisheyeCameraModel::ImgFromCam(
+    const T* params, T u, T v, T w, T* x, T* y) {
   u /= w;
   v /= w;
 
+  // Fisheye coordinates
+  T uu, vv;
+  FisheyeFromNormal(u, v, &uu, &vv);
+
   // Distortion
-  T du, dv;
-  Distortion(&params[4], u, v, &du, &dv);
-  *x = u + du;
-  *y = v + dv;
+  T duu, dvv;
+  Distortion(&params[4], uu, vv, &duu, &dvv);
 
   // Transform to image coordinates
-  *x = f1 * *x + c1;
-  *y = f2 * *y + c2;
+  ImgFromFisheyeCoordinate(params, uu + duu, vv + dvv, x, y);
```

Comment:
nit: ImgFromFisheye and FisheyeFromImg ? To be consistent with NormalFromFisheye and FisheyeFromNormal.